daceml/transformation/constant_folding.py

```diff
@@ -0,0 +1,220 @@
+import copy
+from collections import deque
+from typing import Dict
+
+import numpy as np
+
+import dace
+import dace.data as dt
+from dace.properties import make_properties
+from dace.transformation import pattern_matching
+from dace.sdfg import nodes as nd
+from dace.sdfg import utils as sdutil
+
+import daceml.onnx as donnx
+from daceml.onnx.converters import clean_onnx_name
+from daceml.onnx.nodes.onnx_op import ONNXOp
+from daceml.onnx import ONNXModel
+
+# blocklist of nondeterministic ops
+# yapf: disable
+NONDETERMINISTIC_OPS = {'ONNXGradient',
+                        'ONNXGraphCall',
+                        'ONNXIf',
+                        'ONNXLoop',
+                        'ONNXMomentum',
+                        'ONNXRandomNormal',
+                        'ONNXRandomNormalLike',
+                        'ONNXRandomUniform',
+                        'ONNXRandomUniformLike',
+                        'ONNXSVMClassifier',
+                        'ONNXSVMRegressor',
+                        'ONNXScan',
```

Comment:
is Scan nondeterministic?

Reply:
it takes a graph as an argument, so I guess yes. Didn't read the whole docstring though

Reply:
(the graph can have nondeterministic ops)

Reply:
ok